community/kernel/src/main/java/org/neo4j/unsafe/impl/batchimport/cache/NumberArrayFactory.java

```diff
@@ -212,60 +213,60 @@ public Auto( NumberArrayFactory... candidates )
         @Override
         public LongArray newLongArray( long length, long defaultValue, long base )
         {
-            OutOfMemoryError error = null;
+            Throwable error = null;
             for ( NumberArrayFactory candidate : candidates )
             {
                 try
                 {
                     return candidate.newLongArray( length, defaultValue, base );
                 }
-                catch ( OutOfMemoryError e )
+                catch ( Throwable e )
                 {   // Allright let's try the next one
                     error = e;
                 }
             }
-            throw error( length, 8, error );
+            throw launderedException( error( length, 8, error ) );
```

Comment:
Yes, something like that. In the event of an `OutOfMemoryException` it would be ok to keep the old message, I don't really have a strong opinion about that, just as long as we have some reasonable message for every case (OutOfMemory, Arithmetic, other).